app/services/status-messages.ts

```diff
@@ -0,0 +1,61 @@
+import { service } from '@ember-decorators/service';
+import Service from '@ember/service';
+import config from 'ember-get-config';
+
+export interface StatusMessage {
+    id: string;
+    jumbo?: boolean;
+    dismiss: boolean;
+    extra?: any;
+    class: string;
+}
+
+export default class StatusMessages extends Service {
+    @service cookies;
+
+    messages: StatusMessage[];
+    nextMessages: StatusMessage[];
+
+    constructor() {
+        super();
+        this.nextMessages = this._getCookieMessages();
+    }
+
+    _getCookieMessages(this: StatusMessages): StatusMessage[] {
+        const cookies = this.get('cookies');
+        if (!cookies.read(config.OSF.statusCookie)) {
+            return [];
+        }
+        // decode commas and \\
+        let status = cookies.read(config.OSF.statusCookie).replace(/\\054/g, ',').replace(/\\/g, '');
+        // remove outside quotes
+        status = status.slice(1).slice(0, -1);
+        let messages = JSON.parse(status);
+        messages = messages.map(each => {
+            const updated = each;
+            updated.id = `status.${each.id}`;
+            return updated;
+        });
+        cookies.clear(config.OSF.statusCookie, {
+            domain: config.environment === 'development' ? 'localhost' : '.osf.io',
```

Comment:
Never hardcode the domain. Use a setting in the config that defaults to localhost, i.e. `config.OSF.cookieDomain = 'localhost'`